examples/react-native/__tests__/js/realm-database/schemas/mixed-tests.jsx

```diff
@@ -0,0 +1,145 @@
+import React, {useEffect} from 'react';
+import {Text, View} from 'react-native';
+import {render, waitFor} from '@testing-library/react-native';
+import Realm from 'realm';
+import {createRealmContext} from '@realm/react';
+import Cat from '../../models/Cat';
+
+jest.setTimeout(30000);
+
+const realmConfig = {
+  schema: [Cat],
+  deleteRealmIfMigrationNeeded: true,
+};
+
+const {RealmProvider, useRealm, useQuery} = createRealmContext(realmConfig);
+
+let assertionRealm;
+
+describe('Mixed Tests', () => {
+  beforeEach(async () => {
+    // we will use this Realm for assertions to access Realm Objects outside of a Functional Component (like required by @realm/react)
+    assertionRealm = await Realm.open(realmConfig);
+
+    // delete every object in the realmConfig in the Realm to make test idempotent
+    assertionRealm.write(() => {
+      assertionRealm.delete(assertionRealm.objects(Cat));
+
+      new Cat(assertionRealm, {
+        name: 'Clover',
+        birthDate: new Date('January 21, 2016'),
+      });
+    });
+  });
+  afterAll(() => {
+    if (!assertionRealm.isClosed) {
+      assertionRealm.close();
+    }
+  });
+  it('should create an object with a mixed value', async () => {
+    // :snippet-start: create-mixed-object
+    // :replace-start: {
+    //  "terms": {
+    //   " testID='catItem'": ""
+    //   }
+    // }
+    const CreateCatsInput = () => {
+      const realm = useRealm();
+
+      useEffect(() => {
```

Comment:
do you think it's worth adding a note about what the useEffect hook is doing? 

like i (and i imagine most react devs) would understand that you're adding the cats to the realm when the component mounts, but wonder if it might be adding a comment before the line with smthn like:

```suggestion
      // Add data to the Realm when CreateCatsInput component mounts
      useEffect(() => {
```